packages/patternfly-4/react-charts/src/components/ChartDonut/ChartDonut.js

```diff
@@ -1,21 +1,164 @@
+   */
+  themeVariant: PropTypes.string,
+  /**
+   * The title for the donut chart
+   */
+  title: PropTypes.string,
+  /**
+   * The width props specifies the width of the svg viewBox of the chart container. This value should be given as a
+   * number of pixels.
+   *
+   * Because Victory renders responsive containers, the width and height props do not determine the width and
+   * height of the chart in number of pixels, but instead define an aspect ratio for the chart. The exact number of
+   * pixels will depend on the size of the container the chart is rendered into.
+   *
+   * Note: innerRadius may need to be set when using this property.
+   */
+  width: PropTypes.number
 };
 
+class ChartDonut extends React.Component {
+  titleRef = React.createRef();
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      height: this.props.height || 0,
+      width: this.props.width || 0
+    };
+  }
+
+  componentDidMount() {
+    // Skip resize handler if height and width were provided
+    if (!(this.state.height || this.state.width)) {
+      setTimeout(() => {
+        window.addEventListener('resize', this.handleResize);
+      });
+      this.handleResize();
+    }
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.handleResize);
+  }
+
+  getSubTitle = () => {
+    const { subTitle } = this.props;
+    const { height, width } = this.state;
+
+    if (subTitle && height && width) {
+      const xVal = width / 2 - subTitle.length / 2;
+      const yVal = height / 2 + 10;
+
+      return (
+        <ChartLabel
+          style={styles.label.subTitle}
+          text={subTitle}
+          textAnchor="middle"
+          verticalAnchor="middle"
+          x={xVal}
+          y={yVal}
+        />
+      );
+    }
+    return null;
+  };
+
+  getTitle = () => {
+    const { title } = this.props;
+    const { height, width } = this.state;
+
+    if (title && height && width) {
+      const xVal = width / 2 - title.length / 2;
+      const yVal = height / 2 - 10;
+
+      return (
+        <ChartLabel
+          style={styles.label.title}
+          text={title}
+          textAnchor="middle"
+          verticalAnchor="middle"
+          x={xVal}
+          y={yVal}
+        />
+      );
+    }
+    return null;
+  };
+
+  handleResize = () => {
+    if (this.titleRef.current && this.titleRef.current.parentNode) {
+      this.setState({
+        height: this.titleRef.current.parentNode.clientHeight,
+        width: this.titleRef.current.parentNode.clientWidth
+      });
+    }
+  };
+
+  render() {
+    const { subTitle, theme, themeColor, themeVariant, title, ...rest } = this.props;
+
+    // TODO Replace with PF css variable when available
+    const titleStyles = {
+      height: styles.label.height,
+      position: styles.label.position,
+      width: styles.label.width
+    };
+
+    const chartTheme = theme || getDonutTheme(themeColor, themeVariant);
+
+    return (
+      <React.Fragment>
```

Comment:
I think we are going to have to use the `standalone={false}` prop on the VictoryPie component in this case. That way we can wrap the whole thing in an svg and position the label accordingly. As it is, we are creating 2 separate svg images, then trying to overlay them using `position: absolute`.

Reply:
We can make the api similar to to Victory's by conditionally wrapping the component in an svg based on a `standalone` prop.